command/fmt_test.go

```diff
@@ -18,80 +19,115 @@ func TestFmtCommand(t *testing.T) {
 
 	const inSuffix = ".in.hcl"
 	const expectedSuffix = ".out.hcl"
-	tests := [][]string{
-		{"nomad", "-check"},
-		{"nomad", ""},
-		{"job", "-check"},
-		{"job", ""},
-	}
+
 	tmpDir := t.TempDir()
 
-	for _, test := range tests {
-		t.Run(test[0]+test[1], func(t *testing.T) {
-			inFile := filepath.Join("testdata", "fmt", test[0]+inSuffix)
-			expectedFile := filepath.Join("testdata", "fmt", test[0]+expectedSuffix)
-			fmtFile := filepath.Join(tmpDir, test[0]+".hcl")
+	tests := []struct {
+		name        string
+		testFile    string
+		flags       []string
+		expectWrite bool
+		expectCode  int
+	}{
+		{
+			name:       "config with check",
+			testFile:   "nomad",
+			flags:      []string{"-check"},
+			expectCode: 1,
+		},
+		{
+			name:        "config without check",
+			testFile:    "nomad",
+			flags:       []string{},
+			expectWrite: true,
+			expectCode:  0,
+		},
+		{
+			name:       "job with check",
+			testFile:   "job",
+			flags:      []string{"-check"},
+			expectCode: 1,
+		},
+		{
+			name:        "job without check",
+			testFile:    "job",
+			flags:       []string{},
+			expectWrite: true,
+			expectCode:  0,
+		},
+	}
 
-			input, err := os.ReadFile(inFile)
-			require.NoError(t, err)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			inFile := filepath.Join("testdata", "fmt", tc.testFile+inSuffix)
```

Comment:
Maybe we could parallelize these cases? If so, we can scratch the tmpDir on line 23
```suggestion
			tc := tc
			ci.Parallel(t)
			tmpDir := t.TempDir()
			inFile := filepath.Join("testdata", "fmt", tc.testFile+inSuffix)
```